### PR TITLE
Feat: The scrollbar should fade away when the scroll is finished or not started

### LINKED ISSUE #1169: The scrollbar should fade away when the scroll is finished or not started
When the scroll is not being used, the scrollbar should fade away:

# Current behavior

The scrollbar always stays in view, even when the scroll is finished or not started.

Image

# desired behavior

The scrollbar should fade away when the scroll is finished or not started

https://github.com/twentyhq/twenty/assets/19412894/c5b49b3d-7b18-433b-953f-e22dbe4f6e17

### PR DESCRIPTION
**Pull request by gitstart-twenty**
Demo: 

https://github.com/twentyhq/twenty/assets/140154534/a964fdfc-f1c6-41e2-a346-3f648da47e33

Fixes #1169

### CODE REVIEW

File: front/src/modules/ui/utilities/scroll/hooks/useListenToScroll.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import styled from '@emotion/styled';
+import debounce from 'lodash.debounce';
+
+export const StyledAnimatedScrollBarContainer = styled.div`
+  overflow: auto;
+
+  ::-webkit-scrollbar {
+    width: 0px;
```
Comment (gitstart-twenty) [inline front/src/modules/ui/utilities/scroll/hooks/useListenToScroll.ts:8]: Adding `transition: width 5s ease-in-out;` here appeared to have no effects, also doing this with `framer-motion` is not straightforward.

**Review by Weiko — DISMISSED** [review]

**Review by Weiko — COMMENTED** [review]
The timer starts before we stop scrolling, see video attached. Because of that, the scrollbar is hidden instantly if you scroll for more than 2s, if you stay long enough it flickers (hidding and showing again)


https://github.com/twentyhq/twenty/assets/1834158/01788476-a8c8-4cad-ab47-9ed194d627f1

**Review by charlesBochet — APPROVED** [review]

**Comment by charlesBochet** [comment]
@gitstart-twenty I've made a different implementation taking advantage of native javascript scroll and scrollend events.

**Comment by charlesBochet** [comment]
https://github.com/twentyhq/twenty/assets/12035771/95267121-eb22-4c2e-933f-89015a7c725b

**Comment by gitstart-twenty** [comment]
> @gitstart-twenty I've made a different implementation taking advantage of native javascript scroll and scrollend events.

Awesome

**Pull request merged**